src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/Zaimanhua.kt

```diff
@@ -0,0 +1,212 @@
+        val request = chain.request()
+        if (request.url.host != "v4api.zaimanhua.com" || !request.headers["authorization"].isNullOrBlank()) {
+            return chain.proceed(request)
+        }
+
+        var token: String = preferences.getString("TOKEN", "")!!
+        if (token.isBlank() || !isValid(token)) {
+            val username = preferences.getString("USERNAME", "")!!
+            val password = preferences.getString("PASSWORD", "")!!
+            token = getToken(username, password)
+            if (token.isBlank()) {
+                preferences.edit().putString("TOKEN", "").apply()
+                preferences.edit().putString("USERNAME", "").apply()
+                preferences.edit().putString("PASSWORD", "").apply()
+                return chain.proceed(request)
+            } else {
+                preferences.edit().putString("TOKEN", token).apply()
+                apiHeaders = apiHeaders.newBuilder().setToken(token).build()
+            }
+        }
+        val authRequest = request.newBuilder().apply {
+            header("authorization", "Bearer $token")
+        }.build()
+        return chain.proceed(authRequest)
+    }
+
+    private fun Headers.Builder.setToken(token: String): Headers.Builder = apply {
+        if (token.isNotBlank()) set("authorization", "Bearer $token")
+    }
+
+    private var apiHeaders = headersBuilder().setToken(preferences.getString("TOKEN", "")!!).build()
+
+    private fun isValid(token: String): Boolean {
+        val response = client.newCall(
+            GET(
+                "$accountApiUrl/userInfo/get",
+                headersBuilder().setToken(token).build(),
+            ),
+        ).execute().parseAs<ResponseDto<UserDto>>()
+        return response.errno == 0
+    }
+
+    private fun getToken(username: String, password: String): String {
+        if (username.isBlank() || password.isBlank()) return ""
+        val passwordEncoded =
+            MessageDigest.getInstance("MD5").digest(password.toByteArray(Charsets.UTF_8))
+                .joinToString("") { "%02x".format(it) }
+        val formBody: RequestBody = FormBody.Builder().addEncoded("username", username)
+            .addEncoded("passwd", passwordEncoded).build()
+        val response = client.newCall(
+            POST(
+                "$accountApiUrl/login/passwd",
+                headers,
+                formBody,
+            ),
+        ).execute().parseAs<ResponseDto<UserDto>>()
+        return response.data.user?.token ?: ""
+    }
+
+    // Detail
+    // path: "/comic/detail/mangaId"
+    override fun mangaDetailsRequest(manga: SManga): Request =
+        GET("$apiUrl/comic/detail/${manga.url}", apiHeaders)
+
+    override fun mangaDetailsParse(response: Response): SManga = try {
+        response.parseAs<ResponseDto<ResponseDto<MangaDto>>>().data.data.toSManga()
+    } catch (_: Throwable) {
+        throw Exception("获取漫画信息失败，可能需要用户登录")
+    }
```

Comment:
don't catch every exception, if possible check if login is needed based on current response